### PR TITLE
Servantify Cannon Public endpoint

### PR DESCRIPTION
**Pull request by supersven**
## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [ ] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [ ] If HTTP endpoint paths have been added or renamed, the **endpoint / config-flag checklist** (see Wire-employee only backend [wiki page](https://github.com/zinfra/backend-wiki/wiki/Checklists)) has been followed.
 - [ ] If a cassandra schema migration has been added, I ran **`make git-add-cassandra-schema`** to update the cassandra schema documentation.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.
   - [ ] If new config options introduced: added usage description under docs/reference/config-options.md
   - [ ] If new config options introduced: recommended measures to be taken by on-premise instance operators.
   - [ ] If a cassandra schema migration is backwards incompatible (see also [these docs](https://github.com/wireapp/wire-server/blob/develop/docs/developer/cassandra-interaction.md#cassandra-schema-migrations)), measures to be taken by on-premise instance operators are explained.
   - [ ] If a data migration (not schema migration) introduced: measures to be taken by on-premise instance operators.
   - [ ] If public end-points have been changed or added: does nginz need un upgrade?
   - [ ] If internal end-points have been added or changed: which services have to be deployed in a specific order?

### CODE REVIEW

File: libs/wire-api/src/Wire/API/Routes/WebSocket.hs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import Network.Wai.Handler.WebSockets
+import Network.WebSockets
+import Servant.Server hiding (respond)
+import Servant.Server.Internal.Delayed
+import Servant.Server.Internal.RouteResult
+import Servant.Server.Internal.Router
+import Servant.Swagger
+
+-- | A websocket that relates to a 'PendingConnection'
+-- Copied and adjusted from: <https://hackage.haskell.org/package/servant-websockets-2.0.0/docs/Servant-API-WebSocket.html#t:WebSocketPending>
```
Comment (pcapriotti) [inline libs/wire-api/src/Wire/API/Routes/WebSocket.hs:36]:
```suggestion
-- Copied and adapted from: <https://hackage.haskell.org/package/servant-websockets-2.0.0/docs/Servant-API-WebSocket.html#t:WebSocketPending>
```

File: libs/wire-api/src/Wire/API/Routes/WebSocket.hs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                                 & description ?~ "RFC 6455"
+                                 & url .~ URL "https://datatracker.ietf.org/doc/html/rfc6455"
+                             )
+                    )
+           )
+    where
+      resps :: InsOrdHashMap HttpStatusCode (Referenced Data.Swagger.Response)
+      resps =
+        mempty
+          & at 101 ?~ Inline (mempty & description .~ "This endpoint is accessible via the websocket protocol.")
```
Comment (pcapriotti) [inline libs/wire-api/src/Wire/API/Routes/WebSocket.hs:85]: This description feels a bit out of place. Maybe change it to something like "Connection upgraded."?

File: services/cannon/src/Cannon/API/Public.hs [hunk 3]
```diff
@@ -16,62 +16,27 @@
...
-    Just rs -> return rs -- ensure all middlewares ignore RawResponse - see Note [Raw Response]
+publicAPIServer :: ServerT ServantAPI Cannon
+publicAPIServer = streamData
   where
-    status426 = mkStatus 426 "Upgrade Required"
-    wsoptions = Ws.defaultConnectionOptions
+    streamData :: UserId -> ConnId -> Maybe ClientId -> PendingConnection -> Cannon ()
+    streamData userId connId clientId con = do
+      e <- wsenv
+      liftIO $ wsapp (mkKey userId connId) clientId e con
```
Comment (pcapriotti) [inline services/cannon/src/Cannon/API/Public.hs:42]: This should probably be lifted to the top level.

**Review by pcapriotti — APPROVED** [review]
Looks good!

File: libs/wire-api/src/Wire/API/Routes/WebSocket.hs (hunk 2 above)
Comment (supersven) [reply libs/wire-api/src/Wire/API/Routes/WebSocket.hs]: Sounds good. Applied.

**Pull request merged**